src/api/app/controllers/status_messages_controller.rb

```diff
@@ -17,34 +14,27 @@ def show
   end
 
   def create
-    # check permissions
-    unless permissions.status_message_create
-      raise PermissionDeniedError, 'message(s) cannot be created, you have not sufficient permissions'
-    end
-
-    new_messages = Nokogiri::XML(request.raw_post, &:strict).root
-    @messages = []
-    if new_messages.css('message').present?
-      # message(s) are wrapped in outer xml tag 'status_messages'
-      new_messages.css('message').each do |msg|
-        @messages << StatusMessage.create!(message: msg.content, severity: msg['severity'], user: User.session!)
-      end
-    else
-      # TODO: make use of a validator
-      raise CreatingMessagesError, "no message #{new_messages.to_xml}" if new_messages.name != 'message'
-      # just one message, NOT wrapped in outer xml tag 'status_messages'
-      @messages << StatusMessage.create!(message: new_messages.content, severity: new_messages['severity'], user: User.session!)
-    end
-    render :index
+    status_message = StatusMessage.from_xml(validate_status_message)
+
+    authorize status_message
+
+    status_message.save!
+
+    render_ok
   end
 
   def destroy
-    # check permissions
-    unless permissions.status_message_create
-      raise PermissionDeniedError, 'message cannot be deleted, you have not sufficient permissions'
-    end
-
-    StatusMessage.find(params[:id]).delete
+    status_message = StatusMessage.find(params[:id])
+    authorize status_message
+    status_message.delete
     render_ok
   end
+
+  private
+
+  # TODO: make it more robust
+  def validate_status_message
+    Suse::Validator.validate(:status_message, request.raw_post)
```

Comment:
Not quite sure but doesn't this happen automatically in `ApplicationController.validate_xml_request`?

Reply:
just uncomment and run the specs should make us sure

Reply:
I did the test, and commenting the line out makes the test fail. So in the controller layer, there is no. I see that we have it in another controllers and services too, so i would keep it.